build-system/babel-plugins/babel-plugin-transform-stringish-literals/index.js

```diff
@@ -0,0 +1,148 @@
+/**
+ * Copyright 2020 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+const MERGEABLE_TYPES = ['StringLiteral', 'NumericLiteral', 'TemplateLiteral'];
+
+module.exports = function ({types: t}) {
+  const cloneNodes = (nodes) => nodes.map((node) => t.cloneNode(node));
+  const escapeStringForTemplateLiteral = (value) =>
+    String(value).replace(/\x60/g, '\\`');
+  const canMergeBinaryExpression = (binaryExpression) =>
+    MERGEABLE_TYPES.includes(binaryExpression.left.type) &&
+    MERGEABLE_TYPES.includes(binaryExpression.right.type) &&
+    binaryExpression.operator === '+';
+
+  function whichCloneQuasi(clonedQuasis, index) {
+    for (let i = index; i >= 0; i--) {
+      const quasi = clonedQuasis[i];
+      if (quasi !== null) {
+        return index;
+      }
+    }
+  }
+
+  return {
+    name: 'flatten-stringish-literals',
+    visitor: {
+      BinaryExpression: {
+        exit(path) {
```

Comment:
I think we can simplify this a lot using Babel's evaluator: https://astexplorer.net/#/gist/2f15d0fd95e70d31d1eb9314d68d5b94/b6b9de562a4f71911db7585d0564d74aff85bf41

Reply:
Absolutely. Good idea, will address in a followup.